main/reference/zoe-api/zoe-contract-facet.md

```diff
@@ -71,27 +71,30 @@ await zcf.saveIssuer(secondaryIssuer, keyword);
 ```
 
 ## zcf.makeInvitation(offerHandler, description, customProperties?, proposalShape?)
-- **offerHandler**: **ZCFSeat => Object**
+- **offerHandler**: **(seat: ZCFSeat, offerArgs?: CopyRecord) => any**
 - **description**: **String**
 - **customProperties**: **Object** - Optional.
 - **proposalShape**: **Pattern** - Optional.
 - Returns: **Promise&lt;[Invitation](./zoe-data-types.md#invitation)>**
 
-Makes a credible Zoe **Invitation** for a smart contract. Note that **Invitations** are a special case
-of an ERTP **payment**. They are associated with the **invitationIssuer** and its **mint**, which 
-validate and mint **Invitations**. **zcf.makeInvitation()** serves as an interface to
-the **invitation** **mint**.
-
-The **Invitation**'s
-**value** specifies:
+Uses the Zoe **[InvitationIssuer](./zoe-data-types.md#invitationissuer)** to _mint_
+a credible **Invitation** for a smart contract.
+The returned **Invitation**'s **amount** specifies:
 - The specific contract **instance**.
 - The Zoe **installation**.
 - A unique **[Handle](./zoe-data-types.md#handle)**.
 
-The second argument is a required *description* for the **Invitation**,
+**offerHandler** is a required function accepting a **ZCFSeat** and **offerArgs**
+(which will be present if and only if provided to
+[`E(Zoe).offer(...)`](/reference/zoe-api/zoe.md#e-zoe-offer-invitation-proposal-paymentpkeywordrecord-offerargs))
+and returning arbitrary offer results.
+
+**description** is a required string describing the **Invitation**,
 and should include whatever information is needed for a potential recipient of the **Invitation**
-to know what they are getting in the optional *customProperties* argument, which is
-put in the **Invitation**'s **value**.
+to know what they are getting in the optional **customProperties** argument, which is
+put in the **Invitation**'s **amount**.
+Each one should be a string literal that is unique within its contract and
+used only as the argument to make invitations of a particular kind.
```

Comment:
Oh, man, that is not a small topic, and I don't see it touched on anywhere in these docs. The only reference material for it that I can find is in [endo](https://github.com/endojs/endo/blob/f7979e41a7e83db8bce855c92a34cabcc3bcfa3f/packages/patterns/src/types.js), and it is far from adequate as developer documentation.

I would say "a Pattern[ref] describing the required and allowed components of the proposal. Proposals that don't match the pattern will be rejected by Zoe without interacting with the contract."  I won't claim that's adequate, but to say any more would take a week.

Reply:
I still think you need to explicitly say that they are unique to a static call site within this contract. Or better, within this contract's source text. Otherwise, someone might misread to think they need to be unique per dynamic call to `makeInstance`.

Reply:
@erights's comment applies to the description

Reply:
Yes. Only the description string. Thx.